lib/utils/job-utils/catalog-searcher.js

```diff
@@ -50,8 +52,140 @@ function searchCatalogDataFactory(
         return wwid;
     }
 
+    /**
+     * Get driveId catalogs with virtual disk info extention from megaraid-virtual-disks
+     * @param {String} nodeId - node identifier
+     * @param {Object} filter - [optional] The filter witch contains the driveId catalogs identifier
+     *                          Or devName. For example: {'sda': 1, 'sdb': 1, '3': 1}. 
```

Comment:
I don't understand what the number `1` of each property means.
And also I don't see you use the `filter` as object, seems your code use it as array. (driveId.identifier in filter || driveId.devName in filter)